lib/blockmap.js

```diff
@@ -114,6 +114,20 @@ BlockMap.prototype = {
   constructor: BlockMap,
 
   /**
+   * Add a block range to the block map
+   * @param {Number} start
+   * @param {Number} end
+   * @param {String} checksum - hex digest
+   * @returns {BlockMap}
+   */
+  addRange( start, end, checksum ) {
+    this.ranges.push({ checksum, start, end })
```

Comment:
I guess this ought to check that it doesn't overlap with another existing range?
In that case, I guess this could return a Boolean to indicate if the range was added?